# https://github.com/shaniAmare/long_read_tools

```console
docs/data/tools.json:    "Platform": "Python, Cuda",
docs/data/tools.json:    "Platform": "C++, Cuda, Python",
docs/data/tools.json:          "Title": "Accelerated Dynamic Time Warping on GPU for Selective Nanopore Sequencing",
docs/data/tools.json:    "Strengths_weaknesses": "Fast, does not require GPU. Less accurate than Guppy high accuracy mode",
docs/data/tools.json:    "Platform": "C, Cuda",
docs/data/tools.json:          "Title": "GPU-Accelerated and Pipelined Methylation Calling",
docs/data/tools.json:    "Description": "GPU-enhanced basecaller from ONT",
docs/data/tools.json:    "Platform": "Python, Cuda",
docs/data/tools.json:    "Platform": "Python, Cuda, C++",
docs/data/tools.json:    "Platform": "C++, Cuda",
docs/data/tools.json:    "Overall_performance": "r facilitates both GPU base-calling with ONTs Guppy as well as CPU base-calling with DeepNano-blitz",
docs/data/tools.json:    "Description": "Real-time selective sequencing using ONT with GPU basecalling",
docs/data/tools.json:    "Strengths_weaknesses": "Graphical processing unit (GPU) base-calling is used to avoid requirement of substantial computational resources; Able to scale to gigabase sized references",
docs/data/tools.json:    "Platform": "C, Verilog, CUDA",
docs/data/tools.json:    "Description": "SneakySnake:snake: is the first and the only pre-alignment filtering algorithm that works efficiently and fast on modern CPU, FPGA, and GPU architectures. It greatly (by more than two orders of magnitude) expedites sequence alignment calculation for both short and long reads.",
docs/data/tools.json:    "Underlying_algorithms": "The key idea of SneakySnake is to reduce the approximate string matching (ASM) problem to the single net routing (SNR) problem in VLSI chip layout. In the SNR problem, we are interested in only finding the optimal path that connects two terminals with the least routing cost on a special grid layout that contains obstacles. The SneakySnake algorithm quickly solves the SNR problem and uses the found optimal path to decide whether performing sequence alignment is necessary. Reducing the ASM problem into SNR also makes SneakySnake efficient to implement for modern high-performance computing architectures (CPUs, GPUs, and FPGAs).",
docs/data/tools.json:          "Title": "SneakySnake: a fast and accurate universal genome pre-alignment filter for CPUs, GPUs and FPGAs",
docs/data/tools.json:    "Description": "A SV graph analysis on GPUs or in the cloud",
docs/data/tools-table.json:    "Platform": "Python, Cuda",
docs/data/tools-table.json:    "Strengths_weaknesses": "Fast, does not require GPU. Less accurate than Guppy high accuracy mode",
docs/data/tools-table.json:    "Platform": "C++, Cuda, Python",
docs/data/tools-table.json:          "Title": "Accelerated Dynamic Time Warping on GPU for Selective Nanopore Sequencing",
docs/data/tools-table.json:    "Platform": "C, Cuda",
docs/data/tools-table.json:          "Title": "GPU-Accelerated and Pipelined Methylation Calling",
docs/data/tools-table.json:    "Description": "GPU-enhanced basecaller from ONT",
docs/data/tools-table.json:    "Platform": "Python, Cuda",
docs/data/tools-table.json:    "Platform": "Python, Cuda, C++",
docs/data/tools-table.json:    "Platform": "C++, Cuda",
docs/data/tools-table.json:    "Overall_performance": "r facilitates both GPU base-calling with ONTs Guppy as well as CPU base-calling with DeepNano-blitz",
docs/data/tools-table.json:    "Description": "Real-time selective sequencing using ONT with GPU basecalling",
docs/data/tools-table.json:    "Strengths_weaknesses": "Graphical processing unit (GPU) base-calling is used to avoid requirement of substantial computational resources; Able to scale to gigabase sized references",
docs/data/tools-table.json:    "Platform": "C, Verilog, CUDA",
docs/data/tools-table.json:    "Description": "SneakySnake:snake: is the first and the only pre-alignment filtering algorithm that works efficiently and fast on modern CPU, FPGA, and GPU architectures. It greatly (by more than two orders of magnitude) expedites sequence alignment calculation for both short and long reads.",
docs/data/tools-table.json:    "Underlying_algorithms": "The key idea of SneakySnake is to reduce the approximate string matching (ASM) problem to the single net routing (SNR) problem in VLSI chip layout. In the SNR problem, we are interested in only finding the optimal path that connects two terminals with the least routing cost on a special grid layout that contains obstacles. The SneakySnake algorithm quickly solves the SNR problem and uses the found optimal path to decide whether performing sequence alignment is necessary. Reducing the ASM problem into SNR also makes SneakySnake efficient to implement for modern high-performance computing architectures (CPUs, GPUs, and FPGAs).",
docs/data/tools-table.json:          "Title": "SneakySnake: a fast and accurate universal genome pre-alignment filter for CPUs, GPUs and FPGAs",
docs/data/tools-table.json:    "Description": "A SV graph analysis on GPUs or in the cloud",
docs/data/old_titles.csv:10.1093/bioinformatics/btaa1015,"SneakySnake: a fast and accurate universal genome pre-alignment filter for CPUs, GPUs and FPGAs",2020-11-25
docs/data/titles.csv:10.1093/bioinformatics/btaa1015,"SneakySnake: a fast and accurate universal genome pre-alignment filter for CPUs, GPUs and FPGAs",2020-11-25
docs/data/titles.csv:10.1093/bioadv/vbac088,GPU-Accelerated and Pipelined Methylation Calling,2022-11-29
docs/data/titles.csv:10.1101/2023.03.05.531225,Accelerated Dynamic Time Warping on GPU for Selective Nanopore Sequencing,2023-03-09
long_read_tools_master.csv:CATCaller,10.1101/2020.11.09.374165,"Python, Cuda",An end2end ONT Basecaller using Convolution-augmented Transformer,https://github.com/biomed-AI/CATCaller,NA,Long-Short Range Attention and flattened FFN layer,NA,"Faster than SACall, only trained on bacterial genomes",Generally more accurate than SACall and older Guppy v3.2.2 on specific datasets,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:DeepNano-blitz,10.1093/bioinformatics/btaa297,"Rust, Python",A fast base caller for MinION nanopore sequencers,https://github.com/fmfi-compbio/deepnano-blitz,MIT License,A bi-directional recurrent neural network,NA,"Fast, does not require GPU. Less accurate than Guppy high accuracy mode",NA,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:DTWax,10.1101/2023.03.05.531225,"C++, Cuda, Python",DTWax enables Read Until and yields 1.92X sequencing speedup and 3.64X compute speedup: costup over a sequencing workflow that does not use Read Until on genomes <100kb,https://github.com/harisankarsadasivan/DTWax,Apache 2.0,Nanopolish,NA,NA,NA,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:Guppy,NA,NA,GPU-enhanced basecaller from ONT,Not publicly available,ONT developer license agreement,"flip-flop, RNN",NA,"production basecaller, slower in high-accuracy mode",NA,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:Lokatt,10.1101/2022.07.13.499873,"Python, Cuda, C++",A hybrid DNA nanopore basecaller with an explicit duration hidden Markov model and a residual LSTM network,https://github.com/chunxxc/lokatt,GNU Affero General Public License v3.0,Explicit Duration HMM (EDHMM),NA,trained on E. coli only,92% simplex accuracy,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:Magenta & Maxwell,NA,"C++, Cuda",fast signal-level matching for direct RNA nanopore sequencing,https://github.com/nodrogluap/maxwell,GNU General Public License v2.0,DTW,NA,NA,NA,TRUE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:ReadBouncer ,10.1101/2022.02.01.478636;10.1093/bioinformatics/btac223,"C++, Cmake",A nanopore adaptive sampling tool for Windows and Linux (x64 or ARM64) that uses Interleaved Bloom Filters for live classification of nanopore reads,https://github.com/JensUweUlrich/ReadBouncer,GNU General Public License v3.0,Read Until functionality to unblock reads that match to a given reference sequence database,NA,NA,r facilitates both GPU base-calling with ONTs Guppy as well as CPU base-calling with DeepNano-blitz,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:ReadFish,10.1038/s41587-020-00746-x,Python,Real-time selective sequencing using ONT with GPU basecalling,https://github.com/looselab/readfish,NA,"Package integrates with the Read Until API. However, a slightly modified version of Read Until API V2, is used.",NA,Graphical processing unit (GPU) base-calling is used to avoid requirement of substantial computational resources; Able to scale to gigabase sized references,NA,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:SneakySnake,10.1093/bioinformatics/btaa1015,"C, Verilog, CUDA","SneakySnake:snake: is the first and the only pre-alignment filtering algorithm that works efficiently and fast on modern CPU, FPGA, and GPU architectures. It greatly (by more than two orders of magnitude) expedites sequence alignment calculation for both short and long reads.",https://github.com/CMU-SAFARI/SneakySnake,GNU General Public License v3.0,"The key idea of SneakySnake is to reduce the approximate string matching (ASM) problem to the single net routing (SNR) problem in VLSI chip layout. In the SNR problem, we are interested in only finding the optimal path that connects two terminals with the least routing cost on a special grid layout that contains obstacles. The SneakySnake algorithm quickly solves the SNR problem and uses the found optimal path to decide whether performing sequence alignment is necessary. Reducing the ASM problem into SNR also makes SneakySnake efficient to implement for modern high-performance computing architectures (CPUs, GPUs, and FPGAs).",NA,NA,"SneakySnake significantly improves the accuracy of pre-alignment filtering by up to four orders of magnitude compared to the state-of-the-art pre-alignment filters, Shouji, GateKeeper, and SHD. Using short sequences, SneakySnake accelerates Edlib (state-of-the-art implementation of Myers bit-vector algorithm) and Parasail (sequence aligner with configurable scoring function), by up to 37.7x and 43.9x (>12x on average), respectively, without requiring hardware acceleration, and by up to 413x and 689x (>400x on average), respectively, using hardware acceleration. Using long sequences, SneakySnake accelerates Parasail by up to 979x (140.1x on average). SneakySnake also accelerates the sequence alignment of minimap2, a state-of-the-art read mapper, by up to 6.83x (4.67x on average) and 64.1x (17.1x on average) using short and long sequences, respectively, and without requiring hardware acceleration. As SneakySnake does not replace sequence alignment, users can still configure the aligner of their choice for different scoring functions, surpassing most existing efforts that aim to accelerate sequence alignment.",TRUE,TRUE,TRUE,TRUE,TRUE,FALSE,FALSE,TRUE,FALSE,TRUE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,TRUE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:super-minityper,NA,Python,A SV graph analysis on GPUs or in the cloud,https://github.com/NCBI-Codeathons/super-minityper,MIT License,Uses minimap2 etc for alignment,NA,NA,NA,TRUE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE
long_read_tools_master.csv:HycDemux,10.1186/s13059-023-03053-1,"Python, Cuda",unsupervised approach for accurate barcoded sample demultiplexing in nanopore sequencing,https://github.com/junhaiqi/Hybrid_clustering,GPL-3.0 Licence,NA,NA,NA,NA,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,TRUE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE,FALSE

```
